### PR TITLE
feat: support additional authorized parties

### LINKED ISSUE #231: JWT Authorized Presenter (azp) validation is too strict
Currently, JWT validation throws if the authorized presenter (`azp` field) is not equal to the `client_id` ([see here](https://github.com/panva/node-openid-client/blob/master/lib/client.js#L906)).

However, according to [Google's OIDC docs](https://developers.google.com/identity/protocols/OpenIDConnect#obtainuserinfo), it is specifically different from the `aud` (Audience) field if present (emphasis mine) :

> **`azp`** : The `client_id` of the authorized presenter. This claim is only needed when **the party requesting the ID token is not the same as the audience of the ID token**. This may be the case at Google for hybrid apps where a web application and Android app have a different OAuth 2.0 client_id but share the same Google APIs project.

More specifically, it happens in Android mobile app authentication flow : 

1. The app requests an ID token from Google using the app's cert and client ID, and the target server's client_id
2. Google issues an ID token with `aud` set to the server's client_id and `azp` to the app's client ID
3. The app gets the token and sends it to the server for logging in
4. The server receives the id token, validates it, and issues credentials based on the claims of the id token

AFAICT, this auth flow is currently not possible with node-openid-client

### PR DESCRIPTION
**Pull request by svvac**
I have a variant ready without `allowedThirdParties: true` support if it's a no-no.

Includes #233 which I'm going to close (what was I thinking splitting this?)

Closes #231

### CODE REVIEW

**Comment by panva** [comment]
> I have a variant ready without `allowedThirdParties: true` support if it's a no-no.

It is a no-no as previously discussed.

**Comment by svvac** [comment]
That's sad, but I can live with it.

**Comment by panva** [comment]
> That's sad, but I can live with it.

Okay.

I've pushed a commit to your upstream, please see if i didn't mess up anything.

**Comment by panva** [comment]
May I also ask you to also rename authorizedThirdParties to additionalAuthorizedParties?

**Comment by svvac** [comment]
LGTM

- [x] Renamed the option to `additionalAuthorizedParties`
- [x] Squashed reviews in appropriate commits

File: types/index.d.ts [hunk 1]
```diff
@@ -334,13 +334,17 @@ export interface IntrospectionResponse {
   [key: string]: unknown;
 }
 
+export interface ClientOptions {
+  additionalAuthorizedParties?: false | string | string[];
```
Comment (panva) [inline types/index.d.ts]: Remove the false here. It somehow got back in there.

**Pull request merged**